### DISCUSSION TITLE
Passing user id into modal component

### DISCUSSION THREAD

**Discussion by Chukwusombiri** [wire-elements/modal#286] [General]
Please, is there one who can better explain the reason for getting this error: `Cannot assign string to property App\Http\Livewire\Admin\UserProfile::$user of type App\Models\User` I have a Livewire component for showing my app users and along that line I have a button that opens users details modal. I've passed the user->id into the modal component class. But I cannot see any user when the modal opens except for an error.

**Comment by PhiloNL** [comment]
Set the following public property and allow it to be an int or string depending on your ID type
```
public int|User $user;
```

**Reply by Chukwusombiri** [reply]
Thank you. But its not working. what I have done: I passed user's id(uuid) into the modal and received it on the mount method. I set the public string $user = User::find($user_id) within mount method. in the render method I passed $this->user into a "user" variable. Still not able to get user->name on the blade component. 
`<?php

namespace App\Http\Livewire\Admin;

use App\Models\Contact;
use App\Models\User;
use Livewire\WithFileUploads;
use LivewireUI\Modal\ModalComponent;

class UserProfile extends ModalComponent
{
    public $name;
    public $email;
    public $phone;
    public string $user;

    protected $rules = [
        'name' => ['required', 'string', 'max:255'],
        'email' => ['required', 'string', 'email', 'max:255', 'unique:users'],
        'phone' => ['required',
        'phone:AD,AE,AF,AG,AI,AL,AM,AO,
        AQ,AR,AS,AT,AU,AW,AX,AZ,BA,BB,BD,BE,BF,BG,BH,BI,BJ,
        BL,BM,BN,BO,BQ,BR,BS,BT,BV,BW,BY,BZ,CA,CC,CF,CG,CH,
        CH,CI,CK,CL,CM,CN,CO,CR,CU,CV,CW,CX,CY,CZ,DE,DJ,DK,
        DM,DO,DZ,EC,EE,EG,EH,ER,ES,ET,FI,FJ,FK,FM,FO,FR,GA,
        GB,GE,GD,GF,GG,GH,GI,GL,GM,GN,GP,GQ,GR,GS,GT,GU,GW,
        GY,HK,HM,HN,HR,HT,HU,ID,IE,IL,IM,IN,IO,IQ,IR,IS,IT,
        JE,JM,JO,JP,KE,KG,KH,KI,KM,KN,KP,KR,KW,KY,KZ,LA,LB,
        LC,LI,LK,LR,LS,LT,LU,LV,LY,MA,MC,MD,ME,MF,MG,MH,MK,
        ML,MM,MN,MO,MP,MQ,MR,MS,MT,MU,MV,MW,MX,MY,MZ,NA,NC,
        NE,NF,NG,NI,NL,NO,NR,NU,NZ,OM,PA,PE,PF,PG,PH,PK,PL,
        PN,PR,PS,PT,PW,PY,QA,RE,RN,RO,RS,RU,RW,SA,SB,SC,SD,
        SZ,SE,SG,SH,SI,SJ,SK,SL,SM,SN,SO,SR,SS,ST,SV,SX,SY,
        PM,TC,TD,TF,TG,TH,TJ,TK,TL,TM,TN,TO,TR,TT,TV,TW,TZ,
        UA,UG,UM,US,UY,UZ,VA,VC,VE,VG,VI,VU,WF,YT,ZA,ZM,ZW,
        ',],     
    ];

    public function mount($user){
        $this->user =$user;        
    }

    public static function modalMaxWidth(): string
    {       
        return 'xl';
    }

    public function updated($propertyName)
    {
        $this->validateOnly($propertyName);
    }
   

    public function submit()
    {
       $this->validate();
      
       $user = User::find($this->user);
       $user->name = $this->name;
       $user->email = $this->email;
       $user->phone = $this->phone;      
       $user->save();     
       $this->emit('userUpdated');
    }

    public function render()
    {          
        return view('livewire.admin.user-profile',[
            'user'=>User::find($this->user),
        ]);
    }
}`
